lib/zone.ts

```diff
@@ -1294,7 +1318,13 @@ const Zone: ZoneType = (function(global: any) {
     scheduleMicroTask: scheduleMicroTask,
     showUncaughtError: () => !(Zone as any)[__symbol__('ignoreConsoleErrorUncaughtError')],
     patchEventTargetMethods: () => false,
-    patchOnProperties: noop
+    patchOnProperties: noop,
+    beforeRunTask: (zone: Zone, task: Task) => {
```

Comment:
I see you are making them private, but than we should somehow hide them from the public API.